nautobot/extras/models/jobs.py

```diff
@@ -530,19 +538,59 @@ class JobResult(BaseModel, CustomFieldModel):
         help_text="The object type to which this job result applies",
         on_delete=models.CASCADE,
     )
-    created = models.DateTimeField(auto_now_add=True)
-    completed = models.DateTimeField(null=True, blank=True)
+
+    date_created = models.DateTimeField(auto_now_add=True)
+
+    @property
+    def created(self):
+        return self.date_created
+
+    @created.setter
+    def set_created(self):
+        self.date_created = created
+
+    date_done = models.DateTimeField(null=True, blank=True)
+
+    @property
+    def completed(self):
+        return self.date_done
+
+    @completed.setter
+    def set_completed(self, completed):
+        self.date_done = completed
+
     user = models.ForeignKey(
         to=settings.AUTH_USER_MODEL, on_delete=models.SET_NULL, related_name="+", blank=True, null=True
     )
-    # todoindex:
     status = models.CharField(
         max_length=30,
-        choices=JobResultStatusChoices,
-        default=JobResultStatusChoices.STATUS_PENDING,
+        choices=TaskStateChoices.choices,
+        default=TaskStateChoices.PENDING,
+        help_text="Current state of the Job being run",
+        db_index=True,
     )
     data = models.JSONField(encoder=DjangoJSONEncoder, null=True, blank=True)
-    job_kwargs = models.JSONField(blank=True, null=True, encoder=NautobotKombuJSONEncoder)
+    periodic_task_name = models.CharField(null=True, max_length=255)
+    worker = models.CharField(max_length=100, default=None, null=True)
+    task_args = models.JSONField(blank=True, null=True, encoder=NautobotKombuJSONEncoder)
+    task_kwargs = models.JSONField(blank=True, null=True, encoder=NautobotKombuJSONEncoder)
+    content_type = models.CharField(
+        default="application/x-nautobot-json",
+        max_length=128,
+        verbose_name=('Result Content Type'),
+        help_text='Content type of the result data')
+    content_encoding = models.CharField(
+        default="utf-8",
+        max_length=64,
+        verbose_name=('Result Encoding'),
+        help_text='The encoding used to save the task result data')
```

Comment:
These are currently required by Celery, but since we're always using JSON this is just a passthrough right now. Need to review the base `DatabaseBackend` from `django-celery-results` and see if we can just always use JSON and eliminate or ignore these fields.

Reply:
I still want to see if I can eliminate these fields since they are 100% no longer necessary. This would be done in `JobResultManager.store_result()`.